src/tests/multihost/alltests/test_misc_proxy.py

```diff
@@ -345,10 +351,65 @@ def test_0002_bz1209483(self, multihost, backupsssdconf):
         assert "foo2:*:2001:2001::/home/foo2:/bin/bash" \
                in execute_cmd(multihost, "getent passwd -s "
                                          "sss foo2").stdout_text
-        ssh1 = SSHClient(client_e, username="foo2", password="Secret123")
+        ssh1 = SSHClient(client_e, username="foo2",
+                         password="Secret123")
         ssh1.close()
         execute_cmd(multihost, "userdel -rf foo2")
 
+    def test_bz1927195(self, multihost, backupsssdconf):
+        """
+        :title: sssd runs out of proxy child slots and
+          doesn't clear the counter for Active requests
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1927195
+        :id: 4c5b53e2-ec88-11ec-a41a-845cf3eff344
+        :steps:
+            1. Configure sssd with proxy
+            2. Try sssctl user-checks user@domain 10 times
+            3. Logs should not have error 'All available
+               child slots are full'
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        # When using authentication provider as proxy,
+        # User authentication suddenly stops working and
+        # starts working again only after restarting the sssd service.
+        tools = sssdTools(multihost.client[0])
+        domain_name = tools.get_domain_section_name()
+        domain_params = {'debug_level': '9',
+                         'id_provider': 'ldap',
+                         'proxy_pam_target': 'proxy_sleep',
+                         'auth_provider': 'proxy',
+                         'chpass_provider': 'ldap',
+                         'proxy_max_children': '10',
+                         'enumerate': 'false',
+                         'entry_cache_timeout': '300',
+                         'access_provider': 'proxy'}
+        tools.sssd_conf('domain/' + domain_name, domain_params)
+        tools.clear_sssd_cache()
+        client = multihost.client[0]
+        file_location = "/script/10_ssctl.sh"
+        client.transport.put_file(os.path.dirname(os.path.abspath(__file__))
+                                  + file_location, '/tmp/10_ssctl.sh')
```

Comment:
Will it be better that the for loop is a string instead of using a separate script file? Not insisting, but just trying to avoid having a small file to maintain. The name of the file should be updated to 11_sssctl.sh if you plan to keep it.

Reply:
We have to run all the 11 command at a once to generate 11 parallel requests , will change the script name to 11_sssctl.sh